src/aws_iot_mqtt_client.c

```diff
@@ -169,6 +169,34 @@ IoT_Error_t aws_iot_mqtt_set_connect_params(AWS_IoT_Client *pClient, IoT_Client_
 	FUNC_EXIT_RC(SUCCESS);
 }
 
+IoT_Error_t aws_iot_mqtt_free(AWS_IoT_Client *pClient)
+{
+    IoT_Error_t rc = SUCCESS;
+
+    if (NULL == pClient) {
+        rc = NULL_VALUE_ERROR;
+    }
+
+#ifdef _ENABLE_THREAD_SUPPORT_
+    if (rc == SUCCESS)
+    {
+        rc = aws_iot_thread_mutex_destroy(&(pClient->clientData.state_change_mutex));
+    }
+
+    if (rc == SUCCESS)
```

Comment:
I think we should still attempt to destroy mutexes, even if the previous call to `aws_iot_thread_mutex_destroy` failed.  We could also print a debug message if a call to `aws_iot_thread_mutex_destroy` fails.